### PR TITLE
spdk: allow external connectivity

### PR DESCRIPTION
**Pull request by glimchb**
127.0.0.1 didn't allow us to connect

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

**Comment by glimchb** [comment]
failed CI on
```
spdk_1             | + ./rpc.py nvmf_create_subsystem nqn.2016-06.io.spdk:cnode1 -a -s SPDK00000000000001 -d SPDK_Controller1
spdk_1             | ++ hostname -i
spdk_1             | sh: line 13: hostname: command not found
spdk_1             | + ./rpc.py nvmf_subsystem_add_listener nqn.2016-06.io.spdk:cnode1 -t tcp -a -f ipv4 -s 4444
spdk_1             | usage: rpc.py [options] nvmf_subsystem_add_listener [-h] -t TRTYPE -a TRADDR
spdk_1             |                                                     [-p TGT_NAME] [-f ADRFAM]
spdk_1             |                                                     [-s TRSVCID]
spdk_1             |                                                     nqn
spdk_1             | rpc.py [options] nvmf_subsystem_add_listener: error: argument -a/--traddr: expected one argument
```
investigating now....

File: storage/docker-compose.yml [hunk 1]
```diff
@@ -33,7 +34,7 @@ services:
             ./rpc.py bdev_malloc_create -b Malloc1 64 512 && \
             ./rpc.py nvmf_create_transport -t TCP -u 8192 -m 4 -c 0  && \
             ./rpc.py nvmf_create_subsystem nqn.2016-06.io.spdk:cnode1 -a -s SPDK00000000000001 -d SPDK_Controller1  && \
-            ./rpc.py nvmf_subsystem_add_listener nqn.2016-06.io.spdk:cnode1 -t tcp -a 127.0.0.1 -f ipv4 -s 4444  && \
+            ./rpc.py nvmf_subsystem_add_listener nqn.2016-06.io.spdk:cnode1 -t tcp -a  `hostname -i` -f ipv4 -s 4444  && \
```
Comment (mestery) [inline storage/docker-compose.yml:37]: Wouldn't this be better to just put the docker hostname here? Docker will resolve local DNS entries for you.

**Review by mestery — APPROVED** [review]
My suggestion on the DNS situation would make this a bit cleaner, but fine if you merge either way.

**Comment by glimchb** [comment]
> failed CI on
> 
> ```
> spdk_1             | + ./rpc.py nvmf_create_subsystem nqn.2016-06.io.spdk:cnode1 -a -s SPDK00000000000001 -d SPDK_Controller1
> spdk_1             | ++ hostname -i
> spdk_1             | sh: line 13: hostname: command not found
> spdk_1             | + ./rpc.py nvmf_subsystem_add_listener nqn.2016-06.io.spdk:cnode1 -t tcp -a -f ipv4 -s 4444
> spdk_1             | usage: rpc.py [options] nvmf_subsystem_add_listener [-h] -t TRTYPE -a TRADDR
> spdk_1             |                                                     [-p TGT_NAME] [-f ADRFAM]
> spdk_1             |                                                     [-s TRSVCID]
> spdk_1             |                                                     nqn
> spdk_1             | rpc.py [options] nvmf_subsystem_add_listener: error: argument -a/--traddr: expected one argument
> ```
> 
> investigating now....

of course... multi-stage build... duh...

File: storage/docker-compose.yml (hunk 1 above)
Comment (glimchb) [reply storage/docker-compose.yml:37]: I wanted to, but I can't... SPDK doesn't accept hostnames, only valid IP addresses 
https://spdk.io/doc/jsonrpc.html#nvmf_subsystem_add_listener

Comment (seroyer) [reply storage/docker-compose.yml:37]: Well, you could resolve the IP.  Something like -a `host [the hostname]`

Comment (glimchb) [reply storage/docker-compose.yml:37]: this is what I did above, no ? `-a  `hostname -i` `

**Pull request merged**